Classes/OldModel/Event.php

```diff
@@ -490,23 +508,32 @@ public function hasCities(): bool
      */
     public function getCitiesFromPlaces(): array
     {
-        $cities = [];
-
         // fetch the city name from the corresponding place records
-        $cityData = \Tx_Oelib_Db::selectMultiple(
-            'city',
-            'tx_seminars_sites LEFT JOIN tx_seminars_seminars_place_mm' .
-            ' ON tx_seminars_sites.uid = ' .
-            'tx_seminars_seminars_place_mm.uid_foreign',
-            'uid_local = ' . $this->getUid(),
-            'uid_foreign, city'
-        );
-
-        foreach ($cityData as $city) {
-            $cities[] = $city['city'];
-        }
+        $queryBuilder = self::getQueryBuilderForTable('tx_seminars_sites');
```

Comment:
Please use `getPlacesAsArray` to avoid code duplication. We then can use `array_colum` and `array_unique` on the result.